### ISSUE TITLE
[docs] improve tune distributed tuning guide

### ISSUE THREAD

**Issue by richardliaw** [ray-project/ray#11681]
### Describe your feature request

The distributed tuning guide currently is a bit hard to follow. We probably want to split it into:

1. User guide (overview features)
2. Walkthrough/e2e example.

**Comment by cszhu** [comment]
Hello! This P3 issue has seen no activity in at least a year. It will be closed in 2 weeks as part of ongoing cleanup efforts.

Please remove the `pending-cleanup` label if you believe this issue should remain open.

Thanks for contributing to Ray!
